meta-mender-core/recipes-update-modules/mender-app-update-module/mender-app-update-module_git.bb

```diff
@@ -0,0 +1,60 @@
+DESCRIPTION = "Mender Application Update Module for supporting containerized application updates on devices including related tools."
+
+HOMEPAGE = "https://mender.io"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b4b4cfdaea6d61aa5793b92efd42e081"
+
+SRC_URI = "git://github.com/mendersoftware/app-update-module;branch=master;protocol=https"
+
+SRCREV = "15025d9f1e21eafcbf1325b2a5034b2606e1d719"
+
+S = "${WORKDIR}/git"
+
+RDEPENDS:${PN} = "jq mender-client xdelta3"
+
+PACKAGECONFIG ?= ""
+PACKAGECONFIG[docker-compose] = ",,,docker-compose"
+PACKAGECONFIG[k3s] = ",,,packagegroup-k3s-node"
+PACKAGECONFIG[k8s] = ",,,kubectl"
+
+inherit allarch
+
+do_configure[noexec] = "1"
+do_compile[noexec] = "1"
+
+do_install:class-target() {
+    # install the Application Update Module
```

Comment:
The default `PACKAGECONFIG ?= ""` is empty.
So by default this recipe will just install the `app-update-module` which isn't of any practically useful.
I'd assume this will happen only by accident.

Would add a warning (haven't tested the snippet) to tell them that nothing practical will reach the device unless they specify it:

```
if [[ -n "${PACKAGECONFIG}" ]]; then
   for config_option in ${!PACKAGECONFIG[@]}; do
    options="${options}${config_option}, "
    done

   bbnote warn "You didn't specify a concrete application update type. No specific PACKAGECONFIG set. Options: $options "
fi
````

Reply:
Resolved with:
```
do_compile() {
    if ! ${@bb.utils.contains('PACKAGECONFIG', 'docker-compose', 'true', 'false', d)} && \
       ! ${@bb.utils.contains('PACKAGECONFIG', 'k3s', 'true', 'false', d)} && \
       ! ${@bb.utils.contains('PACKAGECONFIG', 'k8s', 'true', 'false', d)}; then
        bbwarn "PACKAGECONFIG for mender-app-update-module is empty or invalid. Please specify either `docker-compose`, `k3s` or `k8s` unless you know what you are doing."
    fi
}
```